10_rest/runnerdave/pkg/puppy/rest.go

```diff
@@ -0,0 +1,98 @@
+package puppy
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	chi "github.com/go-chi/chi"
+)
+
+// RestStorer store for puppies
+type RestStorer struct {
+	Db Storer
+}
+
+const (
+	getPuppyRoute    = "/api/puppy/{id}"
+	postPuppyRoute   = "/api/puppy/"
+	putPuppyRoute    = "/api/puppy/{id}"
+	deletePuppyRoute = "/api/puppy/{id}"
+)
+
+// GetPuppyRoute Route for the GET method
+func (rs *RestStorer) GetPuppyRoute() string {
+	return getPuppyRoute
+}
+
+// PostPuppyRoute Route for the POST method
+func (rs *RestStorer) PostPuppyRoute() string {
+	return postPuppyRoute
+}
+
+// PutPuppyRoute Route for the PUT method
+func (rs *RestStorer) PutPuppyRoute() string {
+	return putPuppyRoute
+}
+
+// DeletePuppyRoute Route for the PUT method
+func (rs *RestStorer) DeletePuppyRoute() string {
+	return deletePuppyRoute
+}
+
+// GetPuppy HandlerFunction
+func (rs *RestStorer) GetPuppy(w http.ResponseWriter, r *http.Request) {
```

Comment:
All these methods should be private imo.

Reply:
fixed